src/client/canBindGrammar.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ *  Copyright (c) 2021 Red Hat, Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-v20.html
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+
+'use strict';
+
+import { window, workspace, Disposable, TextDocumentContentChangeEvent, TextDocument, Position, commands } from 'vscode';
+
+export function activateCanBindGrammar(bindingWizardProvider: (document: TextDocument, position: Position) => Thenable<boolean>, supportedLanguages: { [id: string]: boolean }): Disposable {
+
+  let disposables: Disposable[] = [];
+  workspace.onDidChangeTextDocument(event => onDidChangeTextDocument(event.document, event.contentChanges), null, disposables);
```

Comment:
I think commands.executeCommand('setContext', 'canBindGrammar', result) must be done only when editor has focus and not when editor content changed

You should remove workspace.onDidChangeTextDocument listener  and compute canbind in 
window.onDidChangeActiveTextEditor(computeCanBindingWithGrammar; null, disposables);

Reply:
What if the user manually types in the code to bind to a grammar?

Reply:
You are right but i find its shame to consume on each changes the canbind lemminx command for this usecase. Perhaps de could skipthis usecase and improvise associategrammarcommand on lemminx side by throwing an exception when dom document is already found with a grammar